### ISSUE TITLE
ilab_s3_can_calculate_srcset results in a mixture of hard and soft crops

### ISSUE THREAD

**Issue by dotherightthing** [Interfacelab/ilab-media-tools#55]
My customised gallery displays a bank of `thumbnail` size images next to a JS driven enlargement. 

Thumbnails are 300 x 300, hard crop. This used to result in all thumbnails being 300 x 300, but lately I have seen some 300 x ??? (soft crop) thumbnails sneaking in.

It appears that the `srcset` contains both the desired `thumbnail` size (hard crop), and an unrelated `medium` size (soft crop).

I can't show you a live example, sorry, as the site can't go live until it is working, and I don't currently have a staging server for PHP 7.

I tried using the Regenerate Thumbnails function which didn't help, but I'm successfully working around this issue by disabling the Media Tools filter:

```
add_filter( 'ilab_s3_can_calculate_srcset', false, 10, 4 );
```

Edit: but note that doing this results in thumbnails being output with bad rotations.

I wondered if this issue might be due to the process at my end:

1. Created 200+ posts with images (days of my trip)
2. Moved images from unpublished posts to a different directory to prevent overloading hosting 
3. Migrated to Media Cloud and AWS/Imgix, without reinstating relocated images first
4. Now editing unpublished posts, reuploading relocated images
5. WordPress / Media Cloud is confused about which properties to use for reuploaded images

**Comment by mushbrain666** [comment]
Regenerate Thumbnails function within ilab media cloud, or the [Regenerate Thumbnails](https://wordpress.org/plugins/regenerate-thumbnails/) wordpress plugin?

**Comment by dotherightthing** [comment]
Thanks, I tried that first and thought it worked, but subsequent runs
didn't help. If there's a newer release of the plugin, I can't test it
until I return from an overseas trip.

**Comment by mushbrain666** [comment]
Are you securing URLs?

Referring to IMGIX signing key, not https

**Comment by dotherightthing** [comment]
Yes I am securing urls, not sure how that works since you share the key
when you share a URL (unless I misunderstand how it works)

**Comment by dotherightthing** [comment]
The Regenerate Thumbnails plugin. Is there a way within the media plugin?

**Comment by mushbrain666** [comment]
Not that I know of.  There is image purge in imgix>tools though I haven't used it. Might be worth looking into if you have an issue with particular image or url. 

With secure URLS you don't share the signing key (called token in this case). What is appended to the URL is a hash of the token and the url - a combination of your unique token and unique URL. This 'secures' the url so nobody can alter it. Presumably if tokens arent secured anybody can alter the url and process an image on your account.

I've been struggling with signing images while doing manual image processing within templates, every change I make to the URL means another hash. You need to sign every unique url. I believe Media Cloud does this for you, but again I'm unsure. 

You can check out docs.imgix.com/setup/securing-images, at the bottom there is a section on securing images within your application and a prompt for client libraries. 

Not sure this is exactly what your looking for, but it seems like an important component of the whole workflow and if it doesnt happen automagically within media cloud you need to do it yourself. 

 "if you need to change the parameters yourself, you'll need to re-sign the new URL."

**Comment by jawngee** [comment]
Srcset has been fixed in 3.0.

Signing happens automatically with Media Cloud.

Regenerate thumbnails is pointless when using Imgix as Imgix is generating thumbnails as needed on the fly.

Can we move more support related questions to:  https://discourse.interfacelab.io/
